code/game/dna/dna_modifier.dm

```diff
@@ -589,6 +589,8 @@
 		var/lock_state = src.connected.locked
 		src.connected.locked = 1//lock it
 
+		nanomanager.update_uis(src) // show the splash while this works
```

Comment:
Actually it's not because the only reason I would ask someone to do that is for performance reasons, and what you are touching modifies performance. So not it's not even similar and frankly I'm embarassed you had to resort to godwin's law desu.